library/ssl_srv.c

```diff
@@ -577,37 +577,59 @@ static int ssl_parse_trusted_ca_keys_ext( mbedtls_ssl_context *ssl,
         own_trusted_auth = ssl->conf->trusted_auth;
         own_key_cert = ssl->conf->key_cert;
 
+        MBEDTLS_SSL_DEBUG_MSG( 3, ( "Trusted CA size %zu", trusted_auth_size ) );
+
+        /* FIXME: Currently this feature is experimental. So don't abort
+         * connection until we are 100% sure, this is really an issue on
+         * the client side. */
         switch( identifier_type )
         {
         case MBEDTLS_SSL_CA_ID_TYPE_PRE_AGREED:
             MBEDTLS_SSL_DEBUG_MSG( 3, ( "Trusted CA type: pre_agreed" ) );
+            if( trusted_auth_size != 0 )
+            {
+                MBEDTLS_SSL_DEBUG_MSG( 1, ( "bad trusted_auth_size" ) );
+                ssl->handshake->trusted_ca_key_cert = NULL;
+                return ( 0 );
+            }
             break;
         case MBEDTLS_SSL_CA_ID_TYPE_KEY_SHA1_HASH:
             MBEDTLS_SSL_DEBUG_MSG( 3, ( "Trusted CA type: key_sha1_hash" ) );
+            if( trusted_auth_size - 1 > 20 )
+            {
+                MBEDTLS_SSL_DEBUG_MSG( 1, ( "bad trusted_auth_size" ) );
+                ssl->handshake->trusted_ca_key_cert = NULL;
+                return ( 0 );
+            }
             break;
         case MBEDTLS_SSL_CA_ID_TYPE_X509_NAME:
             MBEDTLS_SSL_DEBUG_MSG( 3, ( "Trusted CA type: x509_name" ) );
             break;
         case MBEDTLS_SSL_CA_ID_TYPE_CERT_SHA1_HASH:
             MBEDTLS_SSL_DEBUG_MSG( 3, ( "Trusted CA type: cert_sha1_hash" ) );
+            if( trusted_auth_size - 1 > 20 )
+            {
+                MBEDTLS_SSL_DEBUG_MSG( 1, ( "bad trusted_auth_size" ) );
+                ssl->handshake->trusted_ca_key_cert = NULL;
+                return ( 0 );
```

Comment:
return( 0 );

Reply:
space sign should be removed also in the other return() lines

Reply:
Should be fixed now